File: 338/Solution.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ls = [0]*(n+1)
        offset = 0
        for i in range(1, n+1):
            if i < 2**(offset+1):
                ls[i] = 1 + ls[i-(2**offset)]
            else:
                offset += 1
                ls[i] = 1 + ls[i-2**offset] 
        
        return ls
        
        

File: 338/Solution.java
class Solution {
    public int[] countBits(int n) {
        int []x = new int[n+1];
        int offset = 0;
        for(int i=1;i<n+1;i++) {
            if(i < Math.pow(2, offset+1)) {
                x[i] = 1 + x[i - (int)Math.pow(2, offset)];
            }
            else {
                offset++;
                x[i] = 1 + x[i - (int)Math.pow(2, offset)];
            }
            
        }
        return x;
    }
}